### PR TITLE
Add a Policy class

### PR DESCRIPTION
**Pull request by toolness**
As per the discussion in #678, this factors out a new `Policy` class to contain business logic for policies, so that it can easily be reused by different views.

Note that this is only a **refactoring**; it doesn't add any new functionality to the site.

### CODE REVIEW

File: ui/__tests__/util/policy.test.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    id: 1,
+    has_docnode: false,
+    omb_policy_id: 'M-16-19',
+    ...props,
+  });
+}
+
+describe('Policy', () => {
+  it('is frozen', () => {
+    expect(Object.isFrozen(policy())).toBe(true);
```
Comment (toolness) [inline ui/__tests__/util/policy.test.js]: I might be going overboard with this freezing thing, let me know what you think!

File: ui/components/policies/policies-view.js [hunk 2]
```diff
@@ -14,17 +15,15 @@ export default function PoliciesView({ policies, count, topicsIds }) {
       <ThingCounterContainer count={count} singular={singular} plural={plural} />
       <ul className="policy-list list-reset">
         { policies.map(policy =>
-          <Policy key={policy.id} policy={policy} topicsIds={topicsIds} />,
+          <PolicyView key={policy.id} policy={new Policy(policy)} topicsIds={topicsIds} />,
```
Comment (toolness) [inline ui/components/policies/policies-view.js]: I'm rusty on my React, so I'm not sure if constructing a brand-new `Policy` every render is going to irritate React's innards and make it think that props are changing all the time, thus causing lots of extra pointless work. We should figure this out before merging, methinks.

File: ui/util/policy.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  }
+
+  getDocumentLinkProps() {
+    return {
+      route: 'document',
+      params: {
+        policyId: this.omb_policy_id,
+      },
+    };
+  }
```
Comment (toolness) [inline ui/util/policy.js]: These methods *could* be computed properties, but I felt that making them explicit methods might make it easier for readers to distinguish between what are original properties of policy structs vs. extra business logic that we add. Let me know if you disagree though, we can always change them.

**Comment by toolness** [comment]
Um, CircleCI is failing on an API test, so that's confusing. Oh, it looks like it's a recurrence of #673.

File: ui/__tests__/util/policy.test.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  });
+
+  it('copies properties from passed-in object', () => {
+    expect(policy({ id: 4 }).id).toBe(4);
+  });
+
+  it('Serializes (via JSON) to the original passed-in object', () => {
```
Comment (toolness) [inline ui/__tests__/util/policy.test.js]: Honestly there's no strong reason that I added this; I just wanted the `Policy` class to be as close to the struct in behavior as possible, I guess.

File: ui/util/policy.js (hunk 3 above)
Comment (cmc333333) [reply ui/util/policy.js]: I'd actually lean heavier on the properties side; when I see a method call, I assume work is being done. You raise a good point about distinguishing between API values and new logic. Maybe we make the decision about whether or not to duplicate the API's interface and, if not, keep these as methods?

File: ui/util/policy.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+}
+
+Policy.shape = PropTypes.shape({
+  id: PropTypes.number,
+  title_with_number: PropTypes.string,
+  relevant_reqs: PropTypes.number,
+  total_reqs: PropTypes.number,
+  has_docnode: PropTypes.bool,
+  omb_policy_id: PropTypes.string,
+});
```
Comment (cmc333333) [inline ui/util/policy.js]: Neat idea!

**Comment by toolness** [comment]
Cool! Ok, I think your suggestions are good but for now I am gonna do them in separate PRs, and commit this one as-is.

**Pull request merged**